archiver.go

```diff
@@ -1,33 +1,173 @@
 package debos
 
 import (
-	"log"
+	"fmt"
 	"os"
+	"path/filepath"
+	"strings"
 )
 
-func TarOptions(compression string) string {
+type ArchiveType int
+
+// Supported types
+const (
+	_ ArchiveType = iota // Guess archive type from file extension
+	Tar
+	Zip
+	Deb
+)
+
+type Archive struct {
+	File        string // Path to archive file
+	Type        ArchiveType
+	compression string
+}
+
+type Archiver interface {
+	CommandUnpack(destination string, options ...string) []string
+	GetType() ArchiveType
+	SetCompression(compression string) error
+	Unpack(destination string, options ...string) error
+}
+
+func (arc *Archive) SetCompression(compression string) error {
+	return fmt.Errorf("SetCompression is not supported")
+}
+
+func (arc *Archive) GetType() ArchiveType { return arc.Type }
+
+/*
+CommandUnpack construct string for extracting archive file to destination.
+Not intended for direct usage -- mostly for internals and tests.
+*/
+func (arc *Archive) CommandUnpack(destination string, options ...string) []string {
```

Comment:
I'm confused why this needs to be implemented.

Reply:
very useful for unit tests. And to have only one 'Unpack' implementation since the difference is in command line for supported archive types.

Reply:
You can still hvae the central Unpack implementation in your base class even without this fucntion so that's not relevant.

I'll check the usage in unit tests in a bit to see how it gets used, but it seems very odd to me thusfar

Reply:
Yeah, i don't see in the unit test commit how this is useful. It looks like you're only testing explicit implementations so this isn't needed; So the code in here should never be used it can just raise and exception